packages/drivers/odsp-driver/src/createFile.ts

```diff
@@ -70,25 +72,49 @@ export async function createNewFluidFile(
     }
 
     let itemId: string;
+    let content: any;
     let summaryHandle: string = "";
-    let sharingLink: string | undefined;
-    let sharingLinkErrorReason: string | undefined;
+    let shareLinkInfo: ShareLinkInfoType | undefined;
     if (createNewSummary === undefined) {
         itemId = await createNewEmptyFluidFile(
             getStorageToken, newFileInfo, logger, epochTracker, forceAccessTokenViaAuthorizationHeader);
     } else {
-        const content = await createNewFluidFileFromSummary(
+        content = await createNewFluidFileFromSummary(
             getStorageToken,
             newFileInfo,
             logger,
             createNewSummary,
             epochTracker,
             forceAccessTokenViaAuthorizationHeader,
         );
+        debugger;
         itemId = content.itemId;
         summaryHandle = content.id;
-        sharingLink = content.sharingLink;
-        sharingLinkErrorReason = content.sharingLinkErrorReason;
+        const { sharingLink, sharingLinkErrorReason, sharing } = content;
+        if (sharingLink || (sharingLinkErrorReason && !sharing)) {
+            shareLinkInfo = {
+                createLink: {
+                    type: newFileInfo.createLinkType,
+                    link: sharingLink,
+                    error: sharingLinkErrorReason,
+                },
+            };
+        } else if (sharing) {
+            if (sharing.sharingLink) {
+                shareLinkInfo = {
+                    createLink: {
+                        shareId: sharing.shareId,
+                        sharingLink: sharing.sharingLink,
+                    },
+                };
+            } else if (sharing.error) {
+                shareLinkInfo = {
+                    createLink: {
+                        error: sharing.error,
+                    },
+                };
+            }
+        }
```

Comment:
It's hard to say (without traversing more code) what happens if neither condition are met. I.e. what happens or should happen here:
```ts
} else {
???
}
```

A comment would be nice. If this condition should never happen, assert would be nice.

Reply:
Moved the code to a separate function and added description/inline comments to the code to better explain this.